rpmlint/checks/SharedLibraryPolicyCheck.py

```diff
@@ -0,0 +1,129 @@
+from pathlib import Path
+import re
+import stat
+
+import rpm
+from rpmlint.checks.AbstractCheck import AbstractCheck
+from rpmlint.pkg import formatRequire
+from rpmlint.readelfparser import ReadelfParser
+
+
+class SharedLibraryPolicyCheck(AbstractCheck):
+    """
+    Verify shared library packaging policy rules
+
+    This package validates the shared libraries naming scheme based
+    on the Debian/openSUSE shared libraries policy:
+     https://en.opensuse.org/openSUSE:Shared_library_packaging_policy
+     https://www.debian.org/doc/debian-policy/ch-sharedlibs.html
+    """
+
+    def __init__(self, config, output):
+        super().__init__(config, output)
+        self.re_soname_strongly_versioned = re.compile(r'-[\d\.]+\.so$')
+        # the pkgname is based on soname if ending with number; special option is flavor build
+        self.re_soname_pkg = re.compile(r'^lib\S+(\d+(-(32|64)bit)?)$')
+        self.re_so_files = re.compile(r'\S+.so((.(\d+))+)?$')
+
+    def _check_missing_policy_lib(self, pkg):
+        # check the pkg has any libname
+        libfiles = []
+        for fname in pkg.files.keys():
+            if not self.re_so_files.match(fname):
+                continue
+            libfiles.append(fname)
+
+        # if we didn't find any library files then we
+        # don't need to check anything
+        if not libfiles:
+            # verify if name does not match the slpp and if we still don't have any lib then error out
+            if self.re_soname_pkg.match(pkg.name):
+                self.output.add_info('E', pkg, 'shlib-policy-missing-lib')
+
+    def check(self, pkg):
+        if pkg.is_source:
+            return
+
+        # we don't care about develpkgs
```

Comment:
The comment now does not make sense.
Really put in comment in this stuff otherwise nobody will know whats happening later.